src/instamatic/microscope/components/stage.py

```diff
@@ -83,19 +83,19 @@ def set_a_with_speed(self, a: float, speed: int, wait: bool = False):
 
         wait: bool, block until stage movement is complete.
         """
-        with self.rotating_speed(speed):
+        with self.rotation_speed(speed):
             self.set(a=a, wait=False)
         # Do not wait on `set` to return to normal rotation speed quickly
         if wait:
             self.wait()
 
     @contextmanager
-    def rotating_speed(self, speed: int):
+    def rotation_speed(self, speed: Union[float, int]) -> Generator[None, None, None]:
```

Comment:
Use `Number` here as well?

```suggestion
    def rotation_speed(self, speed: Number) -> Generator[None, None, None]:
```